src/main/scala/fi/vm/sade/hakurekisteri/integration/koski/koskiDomain.scala

```diff
@@ -107,12 +106,25 @@ case class KoskiSuoritus(
   tila: Option[KoskiKoodi] = None
 ) {
 
-  def opintopisteitaVahintaan(min: BigDecimal): Boolean = {
-    val sum = osasuoritukset
-      .filter(_.arviointi.exists(_.hyväksytty.contains(true)))
-      .flatMap(_.koulutusmoduuli.laajuus)
-      .map(_.arvo.getOrElse(BigDecimal(0)))
-      .sum
+  def opintopisteitaVahintaan(min: BigDecimal, isOpistovuosi: Boolean = false): Boolean = {
+    val sum =
+      if (isOpistovuosi) {
+        osasuoritukset
+          .map(o =>
+            o.osasuoritukset.get
+              .filter(_.arviointi.exists(_.hyväksytty.contains(true)))
+              .flatMap(_.koulutusmoduuli.laajuus)
+              .map(_.arvo.getOrElse(BigDecimal(0)))
+              .sum
+          )
+          .sum
```

Comment:
Menisiköhän näin, niin tulisi None-keissikin käsiteltyä?

```suggestion
      if (isOpistovuosi) {
        osasuoritukset
          .map(o =>
            o.osasuoritukset
              .map(os =>
                os.filter(_.arviointi.exists(_.hyväksytty.contains(true)))
                  .flatMap(_.koulutusmoduuli.laajuus)
                  .map(_.arvo.getOrElse(BigDecimal(0)))
                  .sum
              )
              .getOrElse(BigDecimal(0))
          )
          .sum
```